WooCommerce/Classes/ViewRelated/Products/ProductSelector/ProductRowViewModel.swift

```diff
@@ -63,6 +63,92 @@ final class ProductRowViewModel: ObservableObject, Identifiable {
     ///
     private let variationDisplayMode: VariationDisplayMode?
 
+    /// Subscription settings extracted from product meta data for a Subscription-type Product, if any
+    ///
+    private(set) var productSubscriptionDetails: ProductSubscription?
+
+    /// Description of the subscription billing details for a Subscription-type Product
+    /// eg: "$60.00 / 2 months"
+    ///
+    var subscriptionBillingDetailsLabel: String {
+        let currency = ServiceLocator.currencySettings.symbol(from: ServiceLocator.currencySettings.currencyCode)
+
+        guard let subscriptionPrice = productSubscriptionDetails?.price,
+              let subscriptionInterval = productSubscriptionDetails?.periodInterval,
+              let subscriptionPeriod = productSubscriptionDetails?.period,
+              let formattedPrice = currencyFormatter.formatAmount(subscriptionPrice, with: currency) else {
+            return ""
+        }
+
+        let subscriptionFrequency = {
+            switch subscriptionInterval {
+            case "1":
+                return subscriptionPeriod.descriptionSingular
+            default:
+                return subscriptionPeriod.descriptionPlural
+            }
+        }
+        return String.localizedStringWithFormat(Localization.formattedProductSubscriptionBilling,
+                                                formattedPrice, subscriptionInterval, subscriptionFrequency())
+    }
+
+    /// Description of the subscription conditions for a Subscription-type Product
+    /// These are separate from each other, a subscription could have any, all, or none
+    /// eg: "$25.00 signup · 1 month free"
+    ///
+    var subscriptionConditionsLabel: String {
+        let currency = ServiceLocator.currencySettings.symbol(from: ServiceLocator.currencySettings.currencyCode)
+
+        // Signup fees
+        guard let signUpFee = productSubscriptionDetails?.signUpFee else {
+            return ""
+        }
+        var formattedSignUpFee: String = ""
+        if signUpFee.isEmpty {
+            formattedSignUpFee = ""
+        } else {
+            formattedSignUpFee = currencyFormatter.formatAmount(signUpFee, with: currency) ?? ""
+        }
+
+        // Trial periods
+        guard let trialLength = productSubscriptionDetails?.trialLength,
+              let trialPeriod = productSubscriptionDetails?.trialPeriod else {
+            return ""
+        }
```

Comment:
Similar comment as above about the use of `guard` check. The implementation would end here if trial length and period are not defined.

Reply:
Yep! As above, updated here: bf632cb9fa90a0ca374561d332ad82e9166466aa , we'll deal with empty values in the formatter at the end of the method better.